membership-attribute-service/app/controllers/AccountController.scala

```diff
@@ -69,28 +71,37 @@ class AccountController extends LazyLogging {
     }
   }
 
-  private def getUpToDatePaymentDetailsFromStripe(defaultPaymentMethodId: Option[String], paymentDetails: PaymentDetails)(implicit tp: TouchpointComponents): Future[PaymentDetails] = {
+  private def findStripeCustomer(customerId: String, likelyStripeService: StripeService)(implicit tp: TouchpointComponents): Future[Option[Stripe.Customer]] = {
+    val alternativeStripeService = if (likelyStripeService == tp.ukStripeService) tp.auStripeService else tp.ukStripeService
+    likelyStripeService.Customer.read(customerId).recoverWith {
+      case _ => alternativeStripeService.Customer.read(customerId)
+    } map(Option(_)) recover {
+      case _ => None
+    }
+  }
+
+  private def getUpToDatePaymentDetailsFromStripe(accountId: Subscription.AccountId, paymentDetails: PaymentDetails)(implicit tp: TouchpointComponents): Future[PaymentDetails] = {
     paymentDetails.paymentMethod.map {
       case card: PaymentCard =>
         (for {
-          paymentMethodId <- OptionT(Future.successful(defaultPaymentMethodId.map(_.trim).filter(_.nonEmpty)))
-          zuoraPaymentMethod <- tp.zuoraService.getPaymentMethod(paymentMethodId).liftM[OptionT]
+          account <- tp.zuoraService.getAccount(accountId).liftM[OptionT]
+          defaultPaymentMethodId <- OptionT(Future.successful(account.defaultPaymentMethodId.map(_.trim).filter(_.nonEmpty)))
+          zuoraPaymentMethod <- tp.zuoraService.getPaymentMethod(defaultPaymentMethodId).liftM[OptionT]
           customerId <- OptionT(Future.successful(zuoraPaymentMethod.secondTokenId.map(_.trim).filter(_.startsWith("cus_"))))
-          maybeUkStripeCustomer <- tp.ukStripeService.Customer.read(customerId).map(Option(_)).recover { case _ => None }.liftM[OptionT]
-          maybeStripeCustomer <-
-            // Performance optimisation not to go to AU Stripe if UK is one gotten above
-            if (maybeUkStripeCustomer.nonEmpty) Future.successful(maybeUkStripeCustomer).liftM[OptionT]
-            else tp.auStripeService.Customer.read(customerId).map(Option(_)).recover { case _ => None }.liftM[OptionT]
+          paymentGateway <- OptionT(Future.successful(account.paymentGateway))
```

Comment:
`OptionT(Future.successful(` and `EitherT(Future.successful(` come up quite a lot in this section of code.  I know it is necessary, but it is quite distracting when reading the code.  A helpful method with a shorter name might make it more readable.